scripts/api.js

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const fs         = require('fs');
+const path       = require('path');
+const yaml       = require('js-yaml');
+const config     = require('../config');
+const sri        = require('./sri.js');
+
+const configFile = config.getConfigPath('_api.yml');
+
+const semver = require('semver');
+
+function getDirectoryNames(source, ignore = []) {
+    return fs.readdirSync(source).map((name) => ignore.indexOf(name) === -1 && name).filter((name) => {
+        if (!name) {
+            return false;
+        }
+        const absolute = path.join(source, name);
+        const stat = fs.lstatSync(absolute);
+
+        // @todo Remove symlink check when twitter-bootstrap files are moved
+        // to the proper directory.
+        return name && (stat.isDirectory() || stat.isSymbolicLink());
+    });
+}
+
+function listFiles(dir) {
+    return fs.readdirSync(dir).reduce((files, file) => {
+        const name = path.join(dir, file);
+        const isDirectory = fs.statSync(name).isDirectory();
+
+        return isDirectory ? [...files, ...listFiles(name)] : [...files, name];
+    }, []);
+}
+
+const assetRegExp = /\.(css|js)$/;
+const cdnPath = path.resolve(__dirname, '../cdn');
+const validSemVer = /^\d+\.\d+\.\d+/;
+
+function generateApiV1(packages) {
+    const endpoint = {};
+
+    console.log('Generating API (v1) endpoints:');
+
+    packages.forEach((name) => {
+        console.log(`/api/v1/${name}`);
+        const packagePath = path.join(cdnPath, name);
+        const versions = semver.rsort(getDirectoryNames(packagePath).filter((v) => validSemVer.test(v)));
+        const tags = {
+            latest: versions[0]
+        };
+        const previous = versions.filter((v) => v[0] !== tags.latest[0])[0] || versions[1] || null;
+
+        if (previous) {
+            tags.previous = previous;
+        }
+
+        endpoint[name] = {
+            tags,
+            versions,
+            assets: {}
+        };
+
+        versions.forEach((version) => {
+            console.log(`/api/v1/${name}/${version}`);
+
+            const assets = {
+                files: []
+            };
+            const versionPath = path.join(packagePath, version);
+            const files = listFiles(versionPath);
+
+            files.forEach((file) => {
+                // Skip files that aren't considered assets.
+                if (!assetRegExp.test(file)) {
+                    return;
+                }
+                const stat = fs.lstatSync(file);
+                const relative = path.relative(versionPath, file);
+                const [, algorithm, hash] = sri.digest(file).split(/(sha\d{3})-(.*)/) || [];
```

Comment:
Currently we don't use Array destructuring anywhere on purpose. Not a big deal, just FYI.

Reply:
This was to avoid creating another (unnecessary) `matches` variable just to extract the indexed values into other variables later. FWIW, the eslint rules prefer object destructuring, I felt this just followed suit.

Reply:
And explicitly disable the rule for Array destructuring :P

Reply:
The `prefer-destructuring` is a preference to enforce the usage of destructuring, not disallow them entirely. I really don't care either way, I just didn't see any lint errors when writing this and thought it was "ok". If you want me to change this I will; I'm not going to nitpick this.